ui/ducks/send/send.js

```diff
@@ -2270,7 +2270,7 @@ export function signTransaction() {
       // merge in the modified txParams. Once the transaction has been modified
       // we can send that to the background to update the transaction in state.
       const unapprovedTxs = getUnapprovedTxs(state);
-      const unapprovedTx = unapprovedTxs[draftTransaction.id];
+      const unapprovedTx = cloneDeep(unapprovedTxs[draftTransaction.id]);
       // We only update the tx params that can be changed via the edit flow UX
```

Comment:
Just curious why it needs deep cloning, can something like destructuring help ?

Reply:
Because transactions have sub objects that also need cloned, not the reference to those sub objects, and if we were to do an object spread it does a shallow copy unless you specify every sub object. Destructuring to get the values we want won't work, because we attach a new key to this object later. A deeper refactor could be done to this for sure but for now this seems like the safest approach to maintain the functionality.

<img width="484" alt="image" src="https://user-images.githubusercontent.com/4448075/220995049-ca14dc73-762c-418f-800f-de0e3ab60cdc.png">